### PR TITLE
[MM-34308] - Migrate 'components/admin_console/group_settings' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by iOSGeekster**
#### Summary
Migrates 'components/admin_console/group_settings' and associated tests to TypeScript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/34308

```release-note
NONE
```

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Review by mickmister — APPROVED** [review]
LGTM :tada: Thanks for the contribution @iOSGeekster!

File: components/admin_console/group_settings/group_settings.tsx [hunk 1]
```diff
@@ -3,23 +3,19 @@
...
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 
 import {getSiteURL} from 'utils/url';
 
-export default class GroupSettings extends React.PureComponent {
-    static propTypes = {
-        isDisabled: PropTypes.oneOfType([
-            PropTypes.func,
-            PropTypes.bool,
-        ]),
```
[the remark this answers is not in the document]
Comment (iOSGeekster) [reply components/admin_console/group_settings/group_settings.tsx:20]: When doing the change, I couldn't find anywhere where a function was passed. But I think it's worth to double-check :-) So I'll do that.

[the remark this answers is not in the document]
Comment (iOSGeekster) [reply components/admin_console/group_settings/group_settings.tsx:20]: Hmm, doens't seem like it. Acutally I cant find many places where `GroupSettings` is used.

**Review by furqanmlk — APPROVED** [review]
No issues found. [E2E](https://automation-dashboard.vercel.app/cycles/bfe301b4-7a80-4902-9d1e-93090420abb2) passed

**Comment by M-ZubairAhmed** [comment]
done in https://github.com/mattermost/mattermost-webapp/pull/10194

**Pull request closed without merging**